File: Dot Net Code/AgroRent/DTOs/UserSignUpDto.cs
using System.ComponentModel.DataAnnotations;

namespace AgroRent.DTOs
{
    public class UserSignUpDto
    {
        [Required]
        [StringLength(20)]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        [StringLength(30)]
        public string LastName { get; set; } = string.Empty;

        [Required]
        [EmailAddress]
        [StringLength(30)]
        public string Email { get; set; } = string.Empty;

        [Required]
        [MinLength(6)]
        public string Password { get; set; } = string.Empty;
    }
}

File: Springboot Code/AgroRent/src/main/java/com/majorproj/agrorent/dto/UserSignUpDto.java
package com.majorproj.agrorent.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class UserSignUpDto {
	@NotBlank(message = "Firtname required")
	private String firstName;
	@NotBlank(message = "LastName required")
	private String lastName;
	@NotBlank(message = "Email required")
	@Email(message = "Inavlid Email")
    private String email;
    @Pattern(regexp = "((?=.*\\d)(?=.*[a-z])(?=.*[#@$*]).{5,20})", message = "invalid password format!!!!")
    private String password;
}
